api/handlers.go

```diff
@@ -39,16 +40,31 @@ type (
 		Storage           int64  `json:"storageLimit"`
 	}
 
-	// userUpdateData defines the fields of the User record that can be changed
+	// accountRecoveryDTO defines the payload we expect when a user is trying to
+	// change their password.
+	accountRecoveryDTO struct {
+		Token           string `json:"token"`
+		Password        string `json:"password"`
+		ConfirmPassword string `json:"confirmPassword"`
+	}
+
+	// credentialsDTO defines the standard credentials package we expect.
+	credentialsDTO struct {
+		Email    string `json:"email"`
+		Password string `json:"password"`
+	}
+
+	// userUpdateDTO defines the fields of the User record that can be changed
 	// externally, e.g. by calling `PUT /user`.
-	userUpdateData struct {
+	userUpdateDTO struct {
 		Email    string `json:"email,omitempty"`
 		StripeID string `json:"stripeCustomerId,omitempty"`
 	}
 )
 
 // healthGET returns the status of the service
 func (api *API) healthGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	defer api.closeBody(req)
```

Comment:
This feels like it should be a middleware on the API that calls this at the end of each handler? Is that a possibility?

Reply:
It is possible, yes. I would strongly advise against it, though. The reason for that is that it's very much non-standard and developers won't expect it, meaning that they will go an close their request bodies and then they'll be surprised by the double-close panics. And all such a middleware would achieve is save us one line per handler. I would rather go for simple, explicit and obvious code.

I don't know how other projects deal with this problem. In `skyd` I believe we just leak those all over the place. It might be a good idea to examine and fix that, if it's really the case.

Reply:
I don't mind if you don't update it, but I disagree it adds complexity. On the contrary, it just ensures it can not be forgotten, if a developer runs into a double close he'll simply notice and realize it's being closed for him by the middleware. But I don't feel strongly so you can resolve this comment.

Reply:
Hey, you are right! I need to go back to that stackoverflow article and leave a comment.

The comment you found was key, I can't believe I haven't seen it. `The Server will close the request body. The ServeHTTP Handler does not need to.`

Here's some code that proves that the body is closed by the server: https://go.dev/play/p/pBuVoF7SyQN
(you will see that it prints ` >>> CLOSED! <<<` on its own).

I'll remove these `Close()` calls.